test/controllers/SeriesDetailsControllerSpec.scala

```diff
@@ -60,42 +65,46 @@ class SeriesDetailsControllerSpec extends FrontEndTestHelper {
     }
 
     "return a redirect to the auth server with an unauthenticated user" in {
-      val controller = new SeriesDetailsController(getUnauthorisedSecurityComponents,
-        new GraphQLConfiguration(app.configuration), getValidKeycloakConfiguration)
+      val graphQLConfiguration = new GraphQLConfiguration(app.configuration)
+      val seriesService = new SeriesService(graphQLConfiguration)
+      val consignmentService = new ConsignmentService(graphQLConfiguration)
+      val controller = new SeriesDetailsController(getUnauthorisedSecurityComponents, getValidKeycloakConfiguration,
+        seriesService, consignmentService)
       val seriesDetailsPage = controller.seriesDetails().apply(FakeRequest(GET, "/series"))
       redirectLocation(seriesDetailsPage).get must startWith ("/auth/realms/tdr/protocol/openid-connect/auth")
       playStatus(seriesDetailsPage) mustBe FOUND
     }
 
-    "render the error page if the api returns errors" in {
+    "render an error if the api returns errors" in {
       val client = new GraphQLConfiguration(app.configuration).getClient[gs.Data, gs.Variables]()
+      val graphQLConfiguration = new GraphQLConfiguration(app.configuration)
+      val seriesService = new SeriesService(graphQLConfiguration)
+      val consignmentService = new ConsignmentService(graphQLConfiguration)
       val data: client.GraphqlData = client.GraphqlData(Option.empty, List(GraphQLClient.Error("Error", Nil, Nil, None)))
       val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
       wiremockServer.stubFor(post(urlEqualTo("/graphql"))
         .willReturn(okJson(dataString)))
 
-      val controller = new SeriesDetailsController(getAuthorisedSecurityComponents,
-        new GraphQLConfiguration(app.configuration), getValidKeycloakConfiguration)
+      val controller = new SeriesDetailsController(getAuthorisedSecurityComponents, getValidKeycloakConfiguration,
+        seriesService, consignmentService)
       val seriesDetailsPage = controller.seriesDetails().apply(FakeRequest(GET, "/series"))
 
-      playStatus(seriesDetailsPage) mustBe BAD_REQUEST
-      contentType(seriesDetailsPage) mustBe Some("text/html")
-      //This test is based on the placeholder error page, it will need to be changed when we implement a new error page.
-      contentAsString(seriesDetailsPage) must include ("Error")
-      contentAsString(seriesDetailsPage) must include ("govuk-error-message")
-
-      wiremockServer.verify(postRequestedFor(urlEqualTo("/graphql")))
+      val failure = seriesDetailsPage.failed.futureValue
+      failure mustBe an[Exception]
```

Comment:
Can this be more specific? I'm guessing if this is a general API error then this should be a GraphQlException

Reply:
You're right. Fixed.